File: core/src/com/mygdx/game/characters/State/ClimbingUp.java
package com.mygdx.game.characters.State;

import com.badlogic.gdx.math.Vector3;
import com.mygdx.game.characters.GameCharacter;
import com.mygdx.game.core.NodeHolder;

/**
 * Created by al on 5/17/2016.
 */
public class ClimbingUp extends State {
    public ClimbingUp(GameCharacter character) {
        super(character);
    }

    @Override
    public void stand() {character.state = new Stand(character);}

    @Override
    public void runLeft() {}

    @Override
    public void runRight() {}

    @Override
    public void walkLeft() {}

    @Override
    public void walkRight() {}

    @Override
    public void climbUp() {
        Vector3 sub = new Vector3(0,-.3f,0);
        sub.rotate(NodeHolder.rotAxis,character.globalRot);
        character.pos.add(sub);
    }

    @Override
    public void climbDown() {character.state = new Stand(character);}
}

File: web/src/characters/state/ClimbingUp.ts
import { State } from './State';
import { Stand } from './Stand';

const CLIMB_SPEED = 0.3;

export class ClimbingUp extends State {
  stand(): void {
    this.character.fsm = new Stand(this.character);
  }
  runLeft(): void {}
  runRight(): void {}
  walkLeft(): void {}
  walkRight(): void {}
  climbUp(): void {
    this.character.addLogicalPos(0, CLIMB_SPEED);
  }
  climbDown(): void {
    this.character.fsm = new Stand(this.character);
  }
}
